### PR TITLE
Revert "Revert "[Asgardeo] Update Documentation to Include Usage of Secrets in Conditional Authentication Scripts""

### PR DESCRIPTION
**Pull request by Sachin-Mamoru**
Reverts wso2/docs-is#4682

### CODE REVIEW

File: en/includes/guides/authentication/conditional-auth/configure-conditional-auth.md [hunk 1]
```diff
@@ -106,4 +106,14 @@ To delete an existing secret:
 
 5. Select the checkbox and confirm your action.
 
-{% endif %}
\ No newline at end of file
+### Using Secrets in Scripts
```
Comment (malithie) [inline en/includes/guides/authentication/conditional-auth/configure-conditional-auth.md:109]: Other headings seems to be in sentence case. So update similarly.

Comment (malithie) [reply en/includes/guides/authentication/conditional-auth/configure-conditional-auth.md:109]: And better heading is termed as 'Use secret in the script'

File: en/includes/guides/authentication/conditional-auth/configure-conditional-auth.md [hunk 2]
```diff
@@ -106,4 +106,14 @@ To delete an existing secret:
...
+
+You may refer to the added secrets in your conditional authentication scripts using the `secrets.key` syntax. For example, to retrieve a secret value, you may use:
```
Comment (malithie) [inline en/includes/guides/authentication/conditional-auth/configure-conditional-auth.md:111]: Adding secrets only talk about adding the secret name and the value. So here you should refer that the user can refer to the secret by using the secret name added in step x. The syntax should be secrets.{secret name}

File: en/includes/references/conditional-auth/api-reference.md [hunk 3]
```diff
@@ -904,13 +907,15 @@ The HTTP GET function enables sending HTTP GET requests to specified endpoints a
             "Accept": "application/json"
         }, authConfig, {
             onSuccess: function(context, data) {
-                Log.info('httpGet call succeeded');
-                context.selectedAcr = data.status;
```
Comment (malithie) [inline en/includes/references/conditional-auth/api-reference.md:908]: why we remove these lines. IMO, this shows how data and context is used. Thus it's a meaningful example

File: en/includes/references/conditional-auth/api-reference.md [hunk 4]
```diff
@@ -965,25 +970,26 @@ The HTTP POST function enables sending HTTP POST requests to specified endpoints
...
         <tr>
-          <td><code>headers</code></td>
+          <td style="white-space: nowrap;"><code>headers</code></td>
           <td>HTTP request headers to be included in the POST request (optional).</td>
         </tr>
         <tr>
-          <td><code>authConfig</code></td>
-          <td>Authentication configuration to be included in the GET request (optional).</td>
+          <td style="white-space: nowrap;"><code>authConfig</code></td>
+          <td>An object containing the necessary metadata to invoke the API. See [AuthConfig](#authconfig) for more information.</td>
```
Comment (malithie) [inline en/includes/references/conditional-auth/api-reference.md:986]: necessary authentication metadata

File: en/includes/references/conditional-auth/api-reference.md [hunk 5]
```diff
@@ -1009,13 +1015,15 @@ The HTTP POST function enables sending HTTP POST requests to specified endpoints
             "Accept": "application/json"
         }, authConfig, {
             onSuccess: function(context, data) {
-                Log.info('httpPost call succeeded');
-                context.selectedAcr = data.status;
```
Comment (malithie) [inline en/includes/references/conditional-auth/api-reference.md:1013]: Refer to my comment on the same functions in httpGet

File: en/includes/references/conditional-auth/api-reference.md [hunk 6]
```diff
@@ -1332,15 +1340,77 @@ It contains the necessary metadata for invoking the API when calling the callCho
   </tr>
 </table>
 
-If the consumer key and the consumer secret are added as secrets, they should be included in the ConnectionMetadata as aliases, as shown below.
+You can securely store consumer keys and secrets as **secrets** in conditional authentication scripts and refer to
+them in your conditional authentication scripts using the `secrets.key` syntax. For example, to retrieve a secret value, you may use:
```
Comment (malithie) [inline en/includes/references/conditional-auth/api-reference.md:1344]: Refer to my comment above

File: en/includes/references/conditional-auth/api-reference.md [hunk 7]
```diff
@@ -1332,15 +1340,77 @@ It contains the necessary metadata for invoking the API when calling the callCho
...
+```angular2html
+var consumerSecret = secrets.clientSecret;
+```
+For more information on adding secrets, refer to the [Add a secret to the script]({{base_path}}/guides/authentication/conditional-auth/configure-conditional-auth/#add-a-secret-to-the-script) section in the 
+documentation.
+
+??? note "Change in behavior from 30th June 2024"
+    Starting from 30th June 2024, you are no longer required to set the aliases for consumer keys and consumer secrets when calling the callChoreo command. Instead, you may directly reference them using the `secrets.secretName` notation.
```
Comment (malithie) [inline en/includes/references/conditional-auth/api-reference.md:1352]: This date should be changed right ?

File: en/includes/references/conditional-auth/api-reference.md [hunk 8]
```diff
@@ -1332,15 +1340,77 @@ It contains the necessary metadata for invoking the API when calling the callCho
...
+        </tr>
+        <tr>
+            <td>bearertoken</td>
+            <td>token</td>
+            <td>Uses a bearer token for authentication.</td>
+        </tr>
+    </tbody>
+</table>
+
+You can securely store sensitive values of properties like username, password, consumerKey, consumerSecret as secrets in conditional authentication scripts and refer to them in your conditional authentication scripts using the `secrets.key` syntax. For example, to retrieve a secret value, you can use:
```
Comment (malithie) [inline en/includes/references/conditional-auth/api-reference.md:1402]: Be consistent on this syntax format. refer to my comment above.

File: en/includes/guides/authentication/conditional-auth/configure-conditional-auth.md (hunk 1 above)
Comment (Sachin-Mamoru) [reply en/includes/guides/authentication/conditional-auth/configure-conditional-auth.md]: Fixed with https://github.com/wso2/docs-is/pull/4773/commits/9acc22f1d66873e2c67cb71939862187674bece9

File: en/includes/guides/authentication/conditional-auth/configure-conditional-auth.md (hunk 2 above)
Comment (Sachin-Mamoru) [reply en/includes/guides/authentication/conditional-auth/configure-conditional-auth.md]: Fixed with https://github.com/wso2/docs-is/pull/4773/commits/9acc22f1d66873e2c67cb71939862187674bece9

File: en/includes/references/conditional-auth/api-reference.md (hunk 3 above)
Comment (Sachin-Mamoru) [reply en/includes/references/conditional-auth/api-reference.md]: Fixed with https://github.com/wso2/docs-is/pull/4773/commits/9acc22f1d66873e2c67cb71939862187674bece9

File: en/includes/references/conditional-auth/api-reference.md (hunk 4 above)
Comment (Sachin-Mamoru) [reply en/includes/references/conditional-auth/api-reference.md]: Fixed with https://github.com/wso2/docs-is/pull/4773/commits/9acc22f1d66873e2c67cb71939862187674bece9

File: en/includes/references/conditional-auth/api-reference.md (hunk 5 above)
Comment (Sachin-Mamoru) [reply en/includes/references/conditional-auth/api-reference.md:1013]: Fixed with https://github.com/wso2/docs-is/pull/4773/commits/9acc22f1d66873e2c67cb71939862187674bece9

File: en/includes/references/conditional-auth/api-reference.md (hunk 6 above)
Comment (Sachin-Mamoru) [reply en/includes/references/conditional-auth/api-reference.md]: Fixed with https://github.com/wso2/docs-is/pull/4773/commits/9acc22f1d66873e2c67cb71939862187674bece9

File: en/includes/references/conditional-auth/api-reference.md (hunk 8 above)
Comment (Sachin-Mamoru) [reply en/includes/references/conditional-auth/api-reference.md]: Fixed with https://github.com/wso2/docs-is/pull/4773/commits/9acc22f1d66873e2c67cb71939862187674bece9

File: en/includes/references/conditional-auth/api-reference.md (hunk 7 above)
Comment (Sachin-Mamoru) [reply en/includes/references/conditional-auth/api-reference.md:1356]: Fixed with https://github.com/wso2/docs-is/pull/4773/commits/9acc22f1d66873e2c67cb71939862187674bece9

File: en/includes/guides/authentication/conditional-auth/configure-conditional-auth.md [hunk 9]
```diff
@@ -79,6 +79,16 @@ To add a new secret:
 
 6. Click **Finish** to complete the creation.
 
+### Use secret in the script
```
Comment (malithie) [inline en/includes/guides/authentication/conditional-auth/configure-conditional-auth.md:82]: In below section ideally we should denote that "This name is what will be used in the script to refer to the secret"
`<tr>
            <td>Secret Name</td>
            <td>A meaningful name for the secret. This name is not changeable.</td>
        </tr>`

File: en/includes/references/conditional-auth/api-reference.md (hunk 7 above)
Comment (malithie) [reply en/includes/references/conditional-auth/api-reference.md:1281]: There's one more entry

Comment (Sachin-Mamoru) [reply en/includes/references/conditional-auth/api-reference.md]: addressed with https://github.com/wso2/docs-is/pull/4773/commits/cc28685065385ccc3454062eccfb6f0d6d5576f9

File: en/includes/guides/authentication/conditional-auth/configure-conditional-auth.md (hunk 9 above)
Comment (Sachin-Mamoru) [reply en/includes/guides/authentication/conditional-auth/configure-conditional-auth.md:82]: addressed with https://github.com/wso2/docs-is/pull/4773/commits/cc28685065385ccc3454062eccfb6f0d6d5576f9

**Review by malithie — APPROVED** [review]

**Pull request merged**